adr-11-centralized-sequencer.md

```diff
@@ -0,0 +1,221 @@
+   }
+   ```
+
+### APIs
+
+The centralized sequencer implements the Generic Sequencer interface from the `go-sequencing` package:
+
+```go
+type Sequencer interface {
+    SubmitRollupBatchTxs(ctx context.Context, req SubmitRollupBatchTxsRequest) (*SubmitRollupBatchTxsResponse, error)
+    GetNextBatch(ctx context.Context, req GetNextBatchRequest) (*GetNextBatchResponse, error)
+    VerifyBatch(ctx context.Context, req VerifyBatchRequest) (*VerifyBatchResponse, error)
+}
+```
+
+These methods are exposed via a gRPC service that rollup clients can interact with.
+The centralized sequencer implements the following methods as part of the Generic Sequencer interface:
+
+1. **SubmitRollupBatchTxs**:
+   - This method is responsible for accepting a batch of transactions from a rollup client. It takes a context and a request containing the rollup ID and the batch of transactions to be submitted.
+   - The method first validates the rollup ID to ensure it matches the expected ID for the sequencer. If the ID is invalid, it returns an error.
+   - Upon successful validation, the method adds the transactions to the internal transaction queue (`TransactionQueue`) for processing.
+   - It then triggers the batch submission process, which involves retrieving the next batch of transactions and submitting them to the designated Data Availability (DA) layer.
+   - Finally, it returns a response indicating the success or failure of the submission.
+
+2. **GetNextBatch**:
+   - This method retrieves the next batch of transactions that are ready to be processed by the rollup. It takes a context and a request containing the rollup ID and the last batch hash.
+   - The method first checks if the rollup ID is valid. If not, it returns an error.
+   - It then verifies the last batch hash to ensure that the rollup client is requesting the correct next batch.
+   - If a valid batch is found, it prepares the batch response, which includes the batch of transactions and a timestamp.
+   - If no transactions are available, it returns an empty batch response.
+
+3. **VerifyBatch**:
+   - This method is used to verify the that a batch received from the sequencer was actually published on the DA layer. It takes a context and a request containing the rollup ID and the batch hash.
+   - Similar to the other methods, it first validates the rollup ID.
+   - It then checks if the provided batch hash exists in the internal data structure that tracks seen batches.
+   - If the batch hash is found, it returns a response indicating that the batch is valid. If not, it returns a response indicating that the batch is invalid.
+
+   This method should be looking at the DA layer to update its view of which batches have been published. 
+
+These methods work together to ensure that the centralized sequencer can effectively manage transaction submissions, retrievals, and verifications, providing a reliable interface for rollup clients to interact with the sequencer.
```

Comment:
should we add something about the batch submission loop?
The centralized sequencer runs a batch submission loop at the batch timer interval which prepares a batch and submits it to DA. The batch timer is configurable.